.github/workflows/rit.yml

```diff
@@ -0,0 +1,73 @@
+name: Rootstock Integration Tests
+
+on:
+  pull_request:
+    types: [opened, reopened]
+    branches:
+      - "**"
+  push:
+    branches:
+      - "master"
+      - "*-rc"
+  workflow_dispatch:
+    inputs:
+      rit-branch:
+        description: 'Branch for Rootstock Integration Tests'
+        required: false
+        default: 'main'
+      powpeg-branch:
+        description: 'Branch for PowPeg Node'
+        required: false
+        default: 'master'
+
+jobs:
+  rootstock-integration-tests:
+    name: Rootstock Integration Tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    steps:
+      - name: Set Branch Variables
+        id: set-branch-variables
+        run: |
+          # Default values
+          RSKJ_BRANCH="master"
+          RIT_BRANCH="${{ github.event.inputs.rit-branch || 'main' }}"
+          POWPEG_BRANCH="${{ github.event.inputs.powpeg-branch || 'master' }}"
+
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            RSKJ_BRANCH="${{ github.event.pull_request.head.ref }}"
+          elif [[ "${{ github.event_name }}" == "push" ]]; then
+            if [[ "${{ github.ref }}" == refs/heads/*-rc ]]; then
+              RSKJ_BRANCH="${{ github.ref }}"
+            fi
+          elif [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
+            RSKJ_BRANCH="${{ github.ref_name }}"
+          fi
+
+          echo "RSKJ_BRANCH=$RSKJ_BRANCH" >> $GITHUB_ENV
+          echo "RIT_BRANCH=$RIT_BRANCH" >> $GITHUB_ENV
+          echo "POWPEG_BRANCH=$POWPEG_BRANCH" >> $GITHUB_ENV
+
+      - name: Checkout rskj Repository
+        if: github.event_name == 'pull_request'
+        uses: actions/checkout@v3
+        with:
+          repository: rsksmart/rskj
+          ref: ${{ env.RSKJ_BRANCH }}
+          fetch-depth: 1
+
+      - name: Checkout Rootstock Integration Tests Repository
+        uses: actions/checkout@v4
+        with:
+          repository: rsksmart/rootstock-integration-tests
+          ref: ${{ env.RIT_BRANCH }}
+          fetch-depth: 0
+
+      - name: Run Rootstock Integration Tests
+        uses: rsksmart/rootstock-integration-tests@v1
```

Comment:
Same comment as above about pinning.
I understand that can be an issue given that the action is publishing always under `v1` tag, but also would be nice to have it pinned.

Reply:
addressed, specified the version `@v1.0.0`

Reply:
I generated a tag with the [full semantic version](https://github.com/rsksmart/rootstock-integration-tests/releases/tag/v1.0.0), if you add it it will work now.

Reply:
I would add the hash instead of the semantic version.

Reply:
I’ve updated the workflow with the commit hash `497172fd38dcfaf48c77f9bb1eeb6617eef5eed6`. You can review these changes in the PowPeg repository, where the workflow has been successfully tested with the hashes. For reference, here is the related job run: [PowPeg Workflow Run](https://github.com/rsksmart/powpeg-node/actions/runs/10692919902)